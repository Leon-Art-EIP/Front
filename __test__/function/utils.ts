import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
