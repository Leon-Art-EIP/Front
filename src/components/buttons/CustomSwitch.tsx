"use client";

import React from "react";
import cn from "classnames";

interface CustomSwitchProps {
  labelOn: string;
  labelOff: string;
  checked: boolean;
  onChange: () => void;
  tooltip?: string;
}

const CustomSwitch: React.FC<CustomSwitchProps> = ({ labelOn, labelOff, checked, onChange, tooltip }) => {
  return (
    <div className="flex items-center cursor-pointer" onClick={onChange} title={tooltip}>
      <span
        className={cn(
          "inline-flex py-1 px-3 text-s rounded-2xl ml-2",
          checked ? "bg-primary text-secondary" : "bg-secondary text-primary",
          "transform translate-y-[-3px]"
        )}
      >
        {checked ? labelOn : labelOff}
      </span>
    </div>
  );
};

export default CustomSwitch;
