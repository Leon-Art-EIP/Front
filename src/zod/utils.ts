export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
