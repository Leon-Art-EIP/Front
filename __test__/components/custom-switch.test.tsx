import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import CustomSwitch from "../../src/components/buttons/CustomSwitch";

test("CrossCircle checked snapshot", () => {
  testSnapshot(<CustomSwitch checked labelOff="Off" labelOn="On" onChange={() => {}} />);
});

test("CrossCircle not checked snapshot", () => {
  testSnapshot(<CustomSwitch checked={false} labelOff="Off" labelOn="On" onChange={() => {}} />);
});

test("CrossCircle checked tooltip snapshot", () => {
  testSnapshot(<CustomSwitch checked labelOff="Off" labelOn="On" onChange={() => {}} tooltip="tooltip" />);
});

test("CrossCircle not checked tooltip snapshot", () => {
  testSnapshot(<CustomSwitch checked={false} labelOff="Off" labelOn="On" onChange={() => {}} tooltip="tooltip" />);
});
