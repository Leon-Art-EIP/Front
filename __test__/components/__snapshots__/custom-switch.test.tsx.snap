// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CrossCircle checked snapshot 1`] = `
<div
  className="flex items-center cursor-pointer"
  onClick={[Function]}
>
  <span
    className="inline-flex py-1 px-3 text-s rounded-2xl ml-2 bg-primary text-secondary transform translate-y-[-3px]"
  >
    On
  </span>
</div>
`;

exports[`CrossCircle checked tooltip snapshot 1`] = `
<div
  className="flex items-center cursor-pointer"
  onClick={[Function]}
  title="tooltip"
>
  <span
    className="inline-flex py-1 px-3 text-s rounded-2xl ml-2 bg-primary text-secondary transform translate-y-[-3px]"
  >
    On
  </span>
</div>
`;

exports[`CrossCircle not checked snapshot 1`] = `
<div
  className="flex items-center cursor-pointer"
  onClick={[Function]}
>
  <span
    className="inline-flex py-1 px-3 text-s rounded-2xl ml-2 bg-secondary text-primary transform translate-y-[-3px]"
  >
    Off
  </span>
</div>
`;

exports[`CrossCircle not checked tooltip snapshot 1`] = `
<div
  className="flex items-center cursor-pointer"
  onClick={[Function]}
  title="tooltip"
>
  <span
    className="inline-flex py-1 px-3 text-s rounded-2xl ml-2 bg-secondary text-primary transform translate-y-[-3px]"
  >
    Off
  </span>
</div>
`;
